### PR TITLE
Adding support for client secret to be passed as a file

### PR DESCRIPTION
**Pull request by CzapBran**
This PR will add support in AzureKeyVaultGlobalConfiguration to initialise the Azure Credentials when a secret File is passed instead of the secret itself.  Requested feature from: 

Fixes https://github.com/jenkinsci/azure-keyvault-plugin/issues/134

- [ X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ X] Ensure that the pull request title represents the desired changelog entry
- [ X] Please describe what you did
- [X ] Link to relevant issues in GitHub or Jira
- [ X] Link to relevant pull requests, esp. upstream and downstream changes
- [ X] Ensure you have provided tests - that demonstrates feature works or fixes the issue

Provided Unit Tests have been added.  I've tested this by supplying both a client secret, a client secret file, and no secret at all

### CODE REVIEW

**Comment by CzapBran** [comment]
One note:

A maven compiler plugin was updated in the pom, not sure if it's necessary

File: pom.xml [hunk 1]
```diff
@@ -170,6 +170,14 @@
...
         </executions>
       </plugin>
+        <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-compiler-plugin</artifactId>
+            <configuration>
+                <source>8</source>
+                <target>8</target>
+            </configuration>
+        </plugin>
```
Comment (timja) [inline pom.xml:180]: remove this it's set in the parent pom

File: src/main/java/org/jenkinsci/plugins/azurekeyvaultplugin/AzureKeyVaultGlobalConfiguration.java [hunk 2]
```diff
@@ -163,6 +167,30 @@ private Optional<String> getPropertyByEnvOrSystemProperty(String envVariable, St
...
     }
 
+    private Optional<String> getPropertyByFile(String envVariable, String systemProperty) {
+        String envResult = System.getenv(envVariable);
+        if (envResult != null) {
+            Path pathToSecret = Paths.get(envResult);
+            try {
+                return Optional.of(Files.readAllLines(pathToSecret).get(0));
+            } catch (IOException e) {
+                throw new RuntimeException(e);
```
Comment (timja) [inline src/main/java/org/jenkinsci/plugins/azurekeyvaultplugin/AzureKeyVaultGlobalConfiguration.java:177]: more appropriate exception
```suggestion
                throw new UncheckedIOException(e);
```

File: src/main/java/org/jenkinsci/plugins/azurekeyvaultplugin/AzureKeyVaultGlobalConfiguration.java [hunk 3]
```diff
@@ -163,6 +167,30 @@ private Optional<String> getPropertyByEnvOrSystemProperty(String envVariable, St
...
+            }
+        }
+
+        String systemResult = System.getProperty(systemProperty);
+        if (systemResult != null) {
+            Path pathToSecret = Paths.get(systemResult);
+            try {
+                return Optional.of(Files.readAllLines(pathToSecret).get(0));
+            } catch (IOException e) {
+                throw new RuntimeException(e);
```
Comment (timja) [inline src/main/java/org/jenkinsci/plugins/azurekeyvaultplugin/AzureKeyVaultGlobalConfiguration.java:187]:
```suggestion
                throw new UncheckedIOException(e);
```

**Review by timja — COMMENTED** [review]
Looks reasonable I assume you've tested this manually as well?

few minor comments, can merge after resolving them

**Comment by CzapBran** [comment]
@timja Yes, tested it locally as well.  Using the following commands:

```
mvn hpi:run -Djenkins.azure-keyvault.url=https://my.vault.azure.net -Djenkins.azure-keyvault.sp.client_id=12345 -Djenkins.azure-keyvault.sp.subscription_id=678 -Djenkins.azure-keyvault.sp.tenant_id=9872 -Djenkins.azure-keyvault.sp.client_secret=123445  
```

The value set in credentials.xml was decoded to be 123445 

Second command was:
```
mvn hpi:run -Djenkins.azure-keyvault.url=https://my.vault.azure.net -Djenkins.azure-keyvault.sp.client_id=12345 -Djenkins.azure-keyvault.sp.subscription_id=678 -Djenkins.azure-keyvault.sp.tenant_id=9872 -Djenkins.azure-keyvault.sp.client_secret_file=work/secrets/secrettext
```

The value on loading the plugin in the credentials.xml file was `Hellothisissecret`, which is the contents of the file in that location

**Review by timja — APPROVED** [review]

**Pull request merged**